_posts/2016-11-09-composer-mailchimp-subscriptions-v3-api.md

```diff
@@ -0,0 +1,163 @@
+
+## Background
+
+To quickly summarize the key takeaways from my previous blog posts [on Composer Manager](/2015/10/15/composing-with-composer-manager.html) and [subscribing users to MailChimp lists using the old API](/2016/01/22/composer-mailchimp-subscriptions.html):
+
+- Composer is a tool for managing PHP libraries that your project depends on.
+- Challenges arise managing project-wide dependencies when custom and contributed modules specify their own unique dependencies.
+- [Composer Manager](https://www.drupal.org/project/composer_manager) is a contributed module for Drupal 7 (and formerly Drupal 6) that addresses these challenges and allows contributed and custom modules to depend on PHP libraries managed via Composer.
+- Using a Composer managed PHP package for the MailChimp API, we can easily subscribe users to MailChimp lists in a Drupal custom module without relying on the Mailchimp module.
+- While the [Mailchimp contributed module](https://www.drupal.org/project/mailchimp) is great, sometimes all you need is a simple, lightweight method for subscribing users to mailing lists.
+
+One important development since my previous posts is that Composer Manager has been deprecated for Drupal 8. [Improvements introduced in Drupal 8.1.0](https://www.drupal.org/project/drupal/releases/8.1.0) allow modules to rely on Composer managed dependencies without the need for the Composer Manager module.
+
+## Implementation
+
+There are a few steps we must take so that we can subscribe users to mailing lists in our custom module. We'll review each of these steps in detail:
+
+- Add the MailChimp API v3.0 PHP library as a dependency of our custom module.
+- Ensure that the library is installed for our project.
+- Properly use the library in our custom module to subscribe users to mailing lists.
+
+### Specify the dependency
+
+ThinkShout maintains the [Mailchimp contributed module]() and we were very excited to see that as part of the effort to ["get Drupal off the island"](http://www.garfieldtech.com/blog/off-the-island-2013) they also released [a PHP library for MailChimp API v3.0](https://packagist.org/packages/thinkshout/mailchimp-api-php).
+
+To use this new library, we must specify it as a dependency of our custom module. We do that in a `composer.json` file that sits in our custom module's root directory and requires that library via the following code:
+
+```json
+{
+  "require": {
+    "thinkshout/mailchimp-api-php": ">=1.0.3"
+  }
+}
+```
+
+### Install the library
+
+Composer is intended for projects and therefore requires a Drupal site to have a single `composer.json`, so things get complicated when individual modules specify their own dependencies.
+
+For Drupal 7 sites (or still active Drupal 6 sites), the [Composer Manager](https://www.drupal.org/project/composer_manager) contributed module handles this by merging all custom and contributed module requirements into a single, consolidated, site-wide `composer.json` file.
```

Comment:
I think it might be helpful to say:
>...handles this by merging all custom and contributed module's requirements and `composer.json` files into a single, consolidated, site-wide `composer.json` file.`

(Did I get that right? That this module takes all the individual composer.json files and merges them into the site-wide one in the root dir?)